src/tomography/statetomography.jl

```diff
@@ -461,42 +446,35 @@ The model can be either a pure state (MPS) or a density operator (LPDO).
  - `test_data`: data for computing cross-validation.
  - `outputpath`: if provided, save metrics on file.
 """
-function tomography(train_data::Matrix{Pair{String, Int}}, L::LPDO;
-                    optimizer::Optimizer,
-                    observer! = nothing,
-                    batchsize::Int64 = 100,
-                    epochs::Int64 = 1000,
-                    kwargs...)
+function tomography(train_data::Matrix{Pair{String, Int}}, L::LPDO; observer! = nothing, kwargs...)
+  
   # Read arguments
-  target = get(kwargs,:target,nothing)
-  test_data = get(kwargs,:test_data,nothing)
-  outputpath = get(kwargs,:fout,nothing)
-  outputlevel = get(kwargs,:outputlevel,1)
+  optimizer::Optimizer         = get(kwargs,:optimizer,SGD(η = 0.01))
+  batchsize::Int64             = get(kwargs,:batchsize,100)
+  epochs::Int64                = get(kwargs,:epochs,1000)
+  measurement_frequency::Int64 = get(kwargs,:measurement_frequency, 1)
+  test_data                    = get(kwargs,:test_data, nothing)
+  outputpath                   = get(kwargs,:fout, nothing)
+  print_metrics                = get(kwargs,:print_metrics, [])
+ 
+  
+  # configure the observer. if no observer is provided, create an empty one
+  observer! = configure!(observer!,optimizer,batchsize,measurement_frequency,train_data,test_data)
```

Comment:
So at first glance the `configure!` and `update!` functionality look fairly complicated to me. I think I understand what you are doing with it, is this something that you need to be looked over carefully or are you happy with the design? My overarching question would be if some of these things like the `measurement_frequency` need to be stored in the `observer!`, but I suppose it makes sense since then the observer has more complete information about what has been measured.

It does make me think that this could be a nice thing to have for DMRG, like the observer could save sweep information so that you can really understand everything about what happened in the calculation from the observer.